sycl/include/CL/sycl/accessor.hpp

```diff
@@ -1219,6 +1225,15 @@ class accessor :
 
   bool operator==(const accessor &Rhs) const { return impl == Rhs.impl; }
   bool operator!=(const accessor &Rhs) const { return !(*this == Rhs); }
+
+private:
+  void checkDeviceAccessorBufferSize(const size_t elemInBuffer) {
+    if (!IsHostBuf && elemInBuffer == 0)
+      throw sycl::runtime_error(
+          "SYCL buffer size is zero. To create a device accessor, SYCL "
+          "buffer size must be greater than zero.",
+          CL_INVALID_BUFFER_SIZE);
```

Comment:
Should we add `PI_INVALID_BUFFER_SIZE = CL_INVALID_BUFFER_SIZE` to the `_pi_result` type and use that here?

Reply:
AFAIK, SYCL exception takes cl_int as an error code (according to SYCL 1.2.1 spec).
https://github.com/intel/llvm/blob/df638c2271d34c1087fd9fc22cc8d0c513e8f53b/sycl/include/CL/sycl/exception.hpp#L47

Besides, I think CL_* would be enough because in this case there's no need to know anything about plugin/backend, thus usage of PI_* can be avoided.
BTW, if you're going to use PI_\*, make sure you have this PI_* for every plugin/backend.